src/test/java/spoon/LauncherTest.java

```diff
@@ -119,4 +121,22 @@ public void testLLauncherBuildModelReturnAModel() {
 
 		assertEquals(2, model.getAllTypes().size());
 	}
+	
+	@Test
+	public void testPrettyPrintWithVirtualFileInput() throws Exception {
+		// contract: prettyPrint() should not throw an exception when used with input from VirtualFile
+		String code = "package foo;\nclass Bar {}\n";
+
+		Launcher launcher = new Launcher();
+		launcher.addInputResource(new VirtualFile(code));
+		launcher.getEnvironment().setNoClasspath(true);
+		launcher.getEnvironment().setAutoImports(true);
+		launcher.getEnvironment().setPreserveLineNumbers(true);
+
+		File tmpDir = Files.createTempDirectory("spoonTestPrettyPrintWithVirtualFileInput").toFile();
+		tmpDir.deleteOnExit();
+		launcher.setSourceOutputDirectory(tmpDir);
```

Comment:
https://spoon.gforge.inria.fr/mvnsites/spoon-core/apidocs/spoon/compiler/Environment.html#setOutputType(spoon.OutputType) Here you can simply set the `OutputType` to none, which is better than creating temp files.